scripts/wallet.js

```diff
@@ -116,66 +220,57 @@ importWallet = async function({
         });
       }
 
-      // Select WIF from internal source OR user input
-      privateKeyForTransactions = newWif || domPrivKey.value;
+      // Select WIF from internal source OR user input (could be: WIF, Mnemonic or xpriv)
+      const privateImportValue = newWif || domPrivKey.value;
       domPrivKey.value = "";
 
-      // Public Key Derivation
-      try {
-        // Incase of an invalid/malformed/incompatible private key: catch and display a nice error!
-        const bArrConvert = from_b58(privateKeyForTransactions);
-        const bArrDropFour = bArrConvert.slice(0, bArrConvert.length - 4);
-        const bKey = bArrDropFour.slice(1, bArrDropFour.length);
-
-        // Extract raw bytes and derive the key from them
-        const pkBytes = bKey.slice(0, bKey.length - 1);
-        publicKeyForNetwork = deriveAddress({
-          pkBytes
-        });
-      } catch (e) {
-        return createAlert('warning', '<b>Failed to import!</b> Invalid private key.' +
-          '<br>Double-check where your key came from!',
-          6000);
+      if (privateImportValue.split(" ").length > 10) {
```

Comment:
I think there should be a slight bit more validation here.

Mnemonics are generally only sets of **12** or **24** words, we could also check with regex for non-alphabetical characters (we'd need to keep in mind international multi-byte languages though, like Japanese).

Note: this is only because I accidently pasted a whole git commit log into the privkey entry, and it did actually attempt to import haha. 😝

Reply:
I changed it to only allow phrases between 12 and 24 words